src/tasks.js

```diff
@@ -0,0 +1,237 @@
+import ProjectStorage from './projectstorage';
+
+const myTodo = [];
+
+function toDo(title, description, date, priority, projectId, done = false) {
+  return {
+    title,
+    description,
+    date,
+    done,
+    priority,
+    projectId,
+  };
+}
+
+function setLocalStorage(data) {
+  localStorage.setItem('myTodo', JSON.stringify(data));
+}
+
+function getLocalStorage() {
+  const ls = localStorage.getItem('myTodo', null, 2);
+
+  return JSON.parse(ls);
+}
+
+
+function checkStorage() {
+  if (getLocalStorage() == null) {
+    return false;
+  }
+
+  return true;
+}
+
+function refreshPage() {
+  window.location.reload();
+}
+
+function saveTodo(todo) {
+  if (checkStorage()) {
+    const ls = getLocalStorage();
+    ls.push(todo);
+    setLocalStorage(ls);
+  } else {
+    myTodo.push(todo);
+    setLocalStorage(myTodo);
+  }
+
+  refreshPage();
+}
+
+function getInputValue() {
+  const titleVal = document.getElementById('title').value;
+  const descVal = document.getElementById('description').value;
+  const dateVal = document.getElementById('date').value;
+  const projectId = document.getElementById('select-project').value;
+  const priorityVal = document.getElementById('priority').value;
+
+  return [titleVal, descVal, dateVal, priorityVal, projectId];
+}
+
+function validate(titleVal, descVal, dateVal, priorityVal, projectId) {
+  if (titleVal === ''
+    || descVal === ''
+    || dateVal === ''
+    || projectId === ''
+    || priorityVal === '') {
+    return false;
+  }
+  return true;
+}
+
+function addTodo() {
+  const [titleVal, descVal, dateVal, priorityVal, projectId] = getInputValue();
+
+  const validateInput = validate(titleVal, descVal, dateVal);
+
+  if (validateInput) {
+    document.querySelector('.bg-modal').style.display = 'none';
+
+    const newTodo = toDo(titleVal, descVal, dateVal, priorityVal, projectId);
+
+    saveTodo(newTodo);
+  }
```

Comment:
DOM-Manipulation and Logic are in separate modules

Reply:
@collinsugwu @abruzy This was not attended to, perhaps it was an oversight